src/meta/src/hummock/manager/mod.rs

```diff
@@ -2484,31 +2509,60 @@ impl HummockManager {
         let window_size = HISTORY_TABLE_INFO_STATISTIC_TIME / (checkpoint_secs as usize);
+
+        let mut group_to_table_vnode_partition = HashMap::default();
+
         for group in &group_infos {
+            let table_vnode_partition_mapping = group_to_table_vnode_partition
+                .entry(group.group_id)
+                .or_insert(BTreeMap::default());
+
             if group.table_statistic.len() == 1 {
+                // no need to handle the dedication compaciton group
                 continue;
             }
 
             for (table_id, table_size) in &group.table_statistic {
-                if !created_tables.contains(table_id) {
-                    continue;
-                }
                 let mut is_high_write_throughput = false;
                 let mut is_low_write_throughput = true;
+                let is_creating_table = !created_tables.contains(table_id);
                 if let Some(history) = table_write_throughput.get(table_id) {
-                    if history.len() >= window_size {
-                        is_high_write_throughput = history.iter().all(|throughput| {
-                            *throughput / checkpoint_secs
-                                > self.env.opts.table_write_throughput_threshold
-                        });
-                        is_low_write_throughput = history.iter().any(|throughput| {
-                            *throughput / checkpoint_secs
-                                < self.env.opts.min_table_split_write_throughput
-                        });
+                    if !is_creating_table {
+                        if history.len() >= window_size {
+                            is_high_write_throughput = history.iter().all(|throughput| {
+                                *throughput / checkpoint_secs
+                                    > self.env.opts.table_write_throughput_threshold
+                            });
+                            is_low_write_throughput = history.iter().any(|throughput| {
+                                *throughput / checkpoint_secs
+                                    < self.env.opts.min_table_split_write_throughput
+                            });
+                        }
+                    } else {
+                        let sum = history.iter().sum::<u64>();
+                        is_low_write_throughput = sum
+                            < self.env.opts.min_table_split_write_throughput
+                                * history.len() as u64
+                                * checkpoint_secs;
                     }
                 }
                 let state_table_size = *table_size;
 
+                {
+                    if !is_low_write_throughput {
+                        table_vnode_partition_mapping.insert(*table_id, 4_u32);
+                    } else if state_table_size > self.env.opts.cut_table_size_limit {
+                        table_vnode_partition_mapping.insert(*table_id, 1_u32);
```

Comment:
Let's avoid hardcoding here. At least we should use a const for the hardcoded number,